product/consul.go

```diff
@@ -15,15 +15,22 @@ const (
 )
 
 // NewConsul takes a product config and creates a Product with all of Consul's default seekers
-func NewConsul(cfg Config) *Product {
-	return &Product{
-		Seekers: ConsulSeekers(cfg.TmpDir, cfg.Since, cfg.Until),
+func NewConsul(cfg Config) (*Product, error) {
+	seekers, err := ConsulSeekers(cfg.TmpDir, cfg.Since, cfg.Until)
+	if err != nil {
+		return nil, err
 	}
+	return &Product{
+		Seekers: seekers,
+	}, nil
 }
 
 // ConsulSeekers seek information about Consul.
-func ConsulSeekers(tmpDir string, since, until time.Time) []*s.Seeker {
-	api := client.NewConsulAPI()
+func ConsulSeekers(tmpDir string, since, until time.Time) ([]*s.Seeker, error) {
+	api, err := client.NewConsulAPI()
```

Comment:
We ought to be passing the `APIClient` into the `<PRODUCT>Seekers` function rather than building it in-place. It will set us up for the future when we migrate ENV var reads to main.go and product-specific TLS fields can be set on `product.Config` in the Agent and passed into products. (💭 And perhaps overridden via .hcl?)

Do you feel that passing the `APIClient` into the `<PRODUCT>Seekers()` fns would be in scope for this set of changes?

Reply:
I completely agree! It feels like it could be slightly outside of scope, and since the PR is already kind of big, I think I'll write up a backlog item for this.

Reply:
I added an item to the internal backlog (CORI-274).